sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -206,11 +169,10 @@ private[hive] class SparkExecuteStatementOperation(
         }
         resultRowSet.addRow(row.toArray.asInstanceOf[Array[Object]])
         curRow += 1
-        resultOffset += 1
       }
-      previousFetchEndOffset = resultOffset
+      val fetchEndOffset = iter.getPosition
       log.info(s"Returning result set with ${curRow} rows from offsets " +
-        s"[$previousFetchStartOffset, $previousFetchEndOffset) with $statementId")
+        s"[$fetchStartOffset, $fetchEndOffset) with $statementId")
```

Comment:
Maybe let's add `getFetchStart` to the interface to be able to just do `[${getFetchStart()}, ${getPosition}]` inline here.